Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,246 @@
+import UIKit
+import SceneKit
+import MapboxMaps
+
+@objc(SceneKitExample)
+
+public class SceneKitExample: UIViewController, ExampleProtocol, CustomLayerHost {
+
+    internal var mapView: MapView!
+    public var peer: MBXPeerWrapper?
+
+    let modelOrigin = CLLocationCoordinate2D(latitude: -35.39847, longitude: 148.9819)
+    var renderer: SCNRenderer!
+    var scene: SCNScene!
+    var modelNode: SCNNode!
+    var cameraNode: SCNNode!
+    var textNode: SCNNode!
+    var useCPUOcclusion = false
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.mapView = MapView(with: view.bounds, resourceOptions: resourceOptions())
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        self.view.addSubview(mapView)
+        mapView.cameraManager.setCamera(
+            centerCoordinate: self.modelOrigin,
+                        zoom: 18,
+                     bearing: 180,
+                       pitch: 60
+        )
+        self.mapView.update { (mapOptions) in
+            mapOptions.render.presentsWithTransaction = true
+        }
+
+        self.mapView.on(.styleLoadingFinished) { [weak self] _ in
+            self?.addModelAndTerrain()
+        }
+    }
+
+    override public func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+         // The below line is used for internal testing purposes only.
+        self.finish()
+    }
+
+    func addModelAndTerrain() {
+        try! mapView.__map.addStyleCustomLayer(forLayerId: "Custom",
+                                                layerHost: self,
+                                            layerPosition: nil)
+
+        var demSource = RasterDemSource()
+        demSource.url = "mapbox://mapbox.mapbox-terrain-dem-v1"
+        demSource.tileSize = 512
+        demSource.maxzoom = 14.0
+        mapView.style.addSource(source: demSource, identifier: "mapbox-dem")
+        let terrain = Terrain(sourceId: "mapbox-dem")
+        _ = self.mapView.style.setTerrain(terrain)
+
+        var skyLayer = SkyLayer(id: "sky-layer")
+        skyLayer.paint?.skyType = .atmosphere
+        skyLayer.paint?.skyAtmosphereSun = .constant([0, 0])
+        skyLayer.paint?.skyAtmosphereSunIntensity = .constant(15.0)
+
+        _ = self.mapView.style.addLayer(layer: skyLayer)
+
+        // Re-use terrain source for hillshade
+        let map = self.mapView.__map!
+        let properties = [
+            "id": "terrain_hillshade",
+            "type": "hillshade",
+            "source": "mapbox-dem",
+            "hillshade-illumination-anchor": "map"
+        ] as [ String: Any ]
+
+        let insertHillshadeBelow = try! map.styleLayerExists(forLayerId: "water") ?
+            LayerPosition(above: nil, below: "water", at: nil) : try! map.styleLayerExists(forLayerId: "hillshade") ?
+            LayerPosition(above: nil, below: "hillshade", at: nil) : nil
```

Comment:
This is a little hard to understand; can we split this into multiple lines.

Reply:
Unnecessary complication - replaced the code with simple `LayerPosition(above: nil, below: "water", at: nil)` as it is present in style used.